apiml-tomcat-common/src/main/java/org/zowe/apiml/filter/AttlsFilter.java

```diff
@@ -32,12 +32,9 @@ public class AttlsFilter extends OncePerRequestFilter {
     @Override
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
         try {
-            if (InboundAttls.getCertificate() != null && InboundAttls.getCertificate().length > 0) {
-                try {
-                    populateRequestWithCertificate(request, InboundAttls.getCertificate());
-                } finally {
-                    InboundAttls.clean();
-                }
+            byte[] certificate = InboundAttls.getCertificate();
+            if (certificate != null && InboundAttls.getCertificate().length > 0) {
```

Comment:
why don't you call length on the certificate?

Reply:
It was there before the change, but the reason is that in case of a missing certificate, an empty array is returned.

Reply:
You are right, the message is always empty. I have refactored the code a little bit (avoid finding the field for each request by reflection). There is no way the exception is thrown (or in the theory low memory etc.). So I guess we do not need to show more information.